bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/handler/capability/HomeCapability.java

```diff
@@ -44,77 +42,67 @@
+        if (hasArea(FeatureArea.SECURITY) && !handler.getCapabilities().containsKey(SecurityCapability.class)) {
             handler.getCapabilities().put(new SecurityCapability(handler));
         }
-        if (hasFeature(FeatureArea.ENERGY) && !handler.getCapabilities().containsKey(EnergyCapability.class)) {
+        if (hasArea(FeatureArea.ENERGY) && !handler.getCapabilities().containsKey(EnergyCapability.class)) {
             handler.getCapabilities().put(new EnergyCapability(handler, descriptionProvider));
         }
         if (firstLaunch) {
             home.getCountry().map(country -> properties.put(PROPERTY_COUNTRY, country));
             home.getTimezone().map(tz -> properties.put(PROPERTY_TIMEZONE, tz));
             properties.put(GROUP_LOCATION, ((Location) home).getLocation().toString());
-            properties.put(PROPERTY_FEATURE, featuresArea.stream().map(f -> f.name()).collect(Collectors.joining(",")));
+            properties.put(PROPERTY_FEATURE,
+                    featureAreas.stream().map(FeatureArea::name).collect(Collectors.joining(",")));
         }
     }
 
     @Override
     protected void afterNewData(@Nullable NAObject newData) {
         super.afterNewData(newData);
-        if (firstLaunch && !hasFeature(FeatureArea.SECURITY)) {
+        if (firstLaunch && !hasArea(FeatureArea.SECURITY)) {
             handler.removeChannels(thing.getChannelsOfGroup(GROUP_SECURITY));
         }
-        if (firstLaunch && !hasFeature(FeatureArea.ENERGY)) {
+        if (firstLaunch && !hasArea(FeatureArea.ENERGY)) {
             handler.removeChannels(thing.getChannelsOfGroup(GROUP_ENERGY));
         }
     }
 
-    private boolean hasFeature(FeatureArea seeked) {
-        return featuresArea.contains(seeked);
-    }
-
-    public NAObjectMap<HomeDataPerson> getPersons() {
-        return persons;
-    }
-
-    public NAObjectMap<HomeDataModule> getModules() {
-        return modules;
+    private boolean hasArea(FeatureArea searched) {
+        return featureAreas.contains(searched);
     }
 
     @Override
     protected List<NAObject> updateReadings(HomeApi api) {
         List<NAObject> result = new ArrayList<>();
-        try {
-            HomeData homeData = api.getHomeData(handler.getId());
-            if (homeData != null) {
-                result.add(homeData);
-                persons = homeData.getPersons();
-                modules = homeData.getModules();
-            }
-            HomeStatus homeStatus = api.getHomeStatus(handler.getId());
-            if (homeStatus != null) {
-                result.add(homeStatus);
-            }
-        } catch (NetatmoException e) {
-            logger.warn("Error getting Home informations : {}", e.getMessage());
-        }
+        handler.getConfiguration().values().stream().filter(String.class::isInstance) //
+                .map(String.class::cast).collect(Collectors.toSet()).forEach(id -> {
```

Comment:
Isn't it a little dangerous and a source of error in the future ? The day we add a new String member to the configuration class, it will be handled as an "id".
I would prefer a loop on securityId, energyId (and id ?)

Reply:
You're correct, this is not the most foolproof thing I ever wrote :)